# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../evms/evms-2.5.5-make-depend.patch
# Copyright (C) 2006 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Submitted by: Christian Wiese (morfoh at opensde.org)
Date: 2006-11-26
Initial Package Version: 2.5.5
Upstream Status: submitted 
Origin: package/filesystem/evms/evms-2.5.5-make-depend.patch by Christian Wiese
Open System Development Environment (http://opensde.org)
Description: Fixes the way evms generates .depend files
While generating the .depend files CPP bails out with: cpp: too many input files

--- evms-2.5.5/make.rules.in.orig	2006-11-26 16:02:13.000000000 +0200
+++ evms-2.5.5/make.rules.in		2006-11-26 16:03:26.000000000 +0200
@@ -137,7 +137,10 @@
 
 # Generating dependencies
 .depend:
-	$(CPP) -M $(EVMS_CPPFLAGS) $(CPPFLAGS) $(EVMS_DEFS) $(SOURCES) > .depend
+	rm -f .depend
+	for i in $(SOURCES); do \
+		$(CPP) -MM $(EVMS_CPPFLAGS) $(CPPFLAGS) $(EVMS_DEFS) $$i >> .depend; \
+	done
 
 # Generating linker version-scripts for the plugins.
 .export:
